ad-joining/register-computer/join.ps1

```diff
@@ -110,5 +158,30 @@ if ($JoinInfo) {
                     "Joining computer to domain failed: $($_.Exception.Message)")
             }
         }
+        finally
+        {
+            if($enableDiagnostics)
```

Comment:
Would it make sense to move the diagnostics code into separate methods?

Reply:
Done.